types.nix

```diff
@@ -235,9 +236,80 @@ rec {
         type = types.attrsOf lvm_vg;
         default = {};
       };
+      nodev = mkOption {
+        type = types.attrsOf (diskoLib.subType { inherit tmpfs; });
+        default = {};
+      };
     };
   };
 
+  tmpfs = types.submodule ({ config, ... }: {
+    options = {
+      type = mkOption {
+        type = types.enum [ "tmpfs" ];
+        internal = true;
+      };
+      size = mkOption {
+        type = types.nullOr types.str; # TODO size type
+        default = null;
+      };
+      mountpoint = mkOption {
+        type = optionTypes.absolute-pathname;
+        default = config._module.args.name;
+      };
+      mountOptions = mkOption {
+        type = types.listOf types.str;
+        default = [];
+      };
+      _meta = mkOption {
+        internal = true;
+        readOnly = true;
+        type = diskoLib.jsonType;
+        default = {
+        };
+      };
+      _create = mkOption {
+        internal = true;
+        readOnly = true;
+        type = types.str;
+        default = "";
+      };
+      _mount = mkOption {
+        internal = true;
+        readOnly = true;
+        type = diskoLib.jsonType;
+        default = {
+          fs.${config.mountpoint} = ''
+            if ! findmnt tmpfs "/mnt${config.mountpoint}" > /dev/null 2>&1; then
+              mount -t tmpfs none "/mnt${config.mountpoint}" \
```

Comment:
```suggestion
              mount -t ${config.type} none "/mnt${config.mountpoint}" \
```

Reply:
hmm, for example for nfs. the none part needs to be the remote. maybe we need an option for the device? I have the feeling, that dedicated types could be better for UX